src/pages/2023-12/next-js-app-router-cache/index.mdx

```diff
@@ -0,0 +1,388 @@
+      <h1>Blog Posts</h1>
+      <ul>
+        {blogData.map((post) => (
+          <li key={post.slug}>
+            <Link href={`/blog/${post.slug}`} prefetch={false}>
+              <a>{post.title}</a>
+            </Link>
+            <p>{post.excerpt}</p>
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+
+The diagram below demonstrates the step-by-step process of how the router cache works.
+
+![router-cache](/articleAssets/2023-12/next-js-app-router-cache/router-cache.png)
+
+#### Cache interactions
+
+Because the full route cache fetches from the data cache, invalidating the data cache means the full route cache gets invalidated too. It now has to re-render components on the server and cache the new render output.
+The inverse is not true, however.
+
+Also, revalidating the data cache does not immediately invalidate the router cache.
+
+#### Build time Caching VS. Run time caching
+
+##### Build time Caching
+
+This refers to caching that occurs during the build process of the application. This type of caching is static and it doesn't change until the next build.
+Full Route Cache stores the RSCP for each page, which is generated at build time. These pages are then served from the cache for each request. As we saw, with this caching, the server does not render these pages for each request.
+
+Build time cache is only updated when the site is rebuilt/redeployed.
+
+##### Run time Caching
+
+Run time caching occurs when the application is accessed by users. It is great for dynamic data and changes based on runtime events and user interactions.
+
+All other caching mechanisms, including Data Cache, Request Memoization and Router Cache fall under this category.
+
+### Closing Notes
+
+While the layered caching model can be difficult to wrap your head around in the beginning, Next.js takes a great approach in aiming to balance the performance of the application, along with data freshness.
+As we saw, all of the caches work in concert but also have distinct roles.
+
+Properly configured, these caching layers allow developers to build highly performant and responsive applications with Next.js, giving users a seamless experience while keeping server and client loads optimized. It's a sophisticated balance of performance, scalability, and maintaining up-to-date content.
+
+And while the official documentation mentions that knowledge of caching is not necessary to be productive with Next.js, it helps a lot to understand its behavior so that you can configure the settings that work best for your particular app.
+
+The table below summarizes all four caching mechanisms and their details.
+
+| Cache               | Description                                                                         | Location | Enabled by default | How to opt-out                                                                                 | Revalidation Criteria                                                                              |
```

Comment:
This chart should be modified slightly.
1. The enabled by default column can be removed.
2. The description should be more focused on what the cache does and why that is important (similar to the purpose column in the Next.js docs). (https://nextjs.org/docs/app/building-your-application/caching#overview)
3. You can also probably remove the opt out section